lib/cocoapods/generator/xcconfig/xcconfig_helper.rb

```diff
@@ -95,12 +101,12 @@ def self.add_settings_for_file_accessors_of_target(target, xcconfig)
         #
         # @return [void]
         #
-        def self.add_static_dependency_build_settings(target, xcconfig, file_accessor)
+        def self.add_static_dependency_build_settings(aggregate_target, pod_target, xcconfig, file_accessor)
           file_accessor.vendored_static_frameworks.each do |vendored_static_framework|
-            XCConfigHelper.add_framework_build_settings(vendored_static_framework, xcconfig, target.sandbox.root)
+            XCConfigHelper.add_framework_build_settings(vendored_static_framework, xcconfig, pod_target.sandbox.root) unless !aggregate_target.nil? && aggregate_target.target_definition.inheritance == 'search_paths'
```

Comment:
nit: `unless !` reads a bit funny to me. could we just replace with `if` and reverse the 2nd condition?